### ISSUE TITLE
Unstable Terraform configuration, keeps switching between two states.

### ISSUE THREAD

**Issue by dgrenner** [airbnb/streamalert#1090]
depends on: #1144 

## Background

I'm using the release-3-0-0 branch, and I am getting a conflict between Terraform resources (I do not know if the same thing happens on master). Every time I run .`/manage.py build` the configuration switches between two states. I get the same result with `terraform apply` - so the issue is not due to the python code changing the Terraform files.

### Description

The output from Terraform is:
```
# module.cloudwatch_monitoring_prod.aws_cloudwatch_metric_alarm.streamalert_lambda_throttles[0] will be updated in-place
  ~ resource "aws_cloudwatch_metric_alarm" "streamalert_lambda_throttles" {
        actions_enabled           = true
      ~ alarm_actions             = [
          + "arn:aws:sns:eu-north-1:123123123123:sr_streamalert_monitoring",
        ]
        ...
      ~ ok_actions                = [
          + "arn:aws:sns:eu-north-1:123123123123:sr_streamalert_monitoring",
        ]
      ~ period                    = 120 -> 300
        ...
      ~ tags                      = {
          - "Cluster" = "prod" -> null
            "Name"    = "StreamAlert"
        }
        ...
    }
```
If I perform the actions, then re-run the command I get:
```
  # module.classifier_prod_lambda.aws_cloudwatch_metric_alarm.lambda_throttles[0] will be updated in-place
  ~ resource "aws_cloudwatch_metric_alarm" "lambda_throttles" {
        actions_enabled           = true
      ~ alarm_actions             = [
          - "arn:aws:sns:eu-north-1:123123123123:sr_streamalert_monitoring",
        ]
        ...
      ~ ok_actions                = [
          - "arn:aws:sns:eu-north-1:123123123123:sr_streamalert_monitoring",
        ]
      ~ period                    = 300 -> 120
        ...
      ~ tags                      = {
          + "Cluster" = "prod"
            "Name"    = "StreamAlert"
        }
        ...
    }
```
So the state keeps switching between these two settings.
The same thing (with same changes to alarm_actions, ok_actions, period and tags) happens with
```
 # module.cloudwatch_monitoring_prod.aws_cloudwatch_metric_alarm.streamalert_lambda_invocation_errors[0] will be updated in-place
  ~ resource "aws_cloudwatch_metric_alarm" "streamalert_lambda_invocation_errors" {
```

So far I've tracked it down to the configuration in two files, `terraform/modules/tf_lambda/cloudwatch.tf` and `terraform/modules/tf_monitoring/main.tf`. Both of them define resources that have the names above. The default period of the one in `tf_lambda` is 120, and the one in `tf_monitoring` is 300 (that is not changed in my configuration files).

As a work-around I tried to set `"cloudwatch_monitoring"/"lambda_alarms_enabled"` in the cluster configuration to false, then I ran manage.py build, then switched to true again. That resulted in that I now have 4 resources that are switching back and forth with every build:
```
  # module.classifier_prod_lambda.aws_cloudwatch_metric_alarm.lambda_invocation_errors[0] will be updated in-place
  ~ resource "aws_cloudwatch_metric_alarm" "lambda_invocation_errors" {
...
  # module.classifier_prod_lambda.aws_cloudwatch_metric_alarm.lambda_throttles[0] will be updated in-place
  ~ resource "aws_cloudwatch_metric_alarm" "lambda_throttles" {
...
  # module.cloudwatch_monitoring_prod.aws_cloudwatch_metric_alarm.streamalert_lambda_invocation_errors[0] will be updated in-place
  ~ resource "aws_cloudwatch_metric_alarm" "streamalert_lambda_invocation_errors" {
...
  # module.cloudwatch_monitoring_prod.aws_cloudwatch_metric_alarm.streamalert_lambda_throttles[0] will be updated in-place
  ~ resource "aws_cloudwatch_metric_alarm" "streamalert_lambda_throttles" {
```

### Steps to Reproduce
The cluster config (if needed for reproducing this) is:
```
{
  "id": "prod",
  "data_sources": {
    "kinesis": {
      "sr_prod_streamalert": [
        "selling_range"
      ]
    }
  },
  "modules": {
    "cloudwatch_logs_destination": {
      "cross_account_ids": [
        "456456456456",
        "123123123123"
      ],
      "regions": [
        "ap-southeast-1",
        "eu-west-1",
        "us-west-2"
      ],
      "enabled": true
    },
    "kinesis": {
      "streams": {
        "retention": 24,
        "shards": 1
      }
    },
    "kinesis_events": {
      "batch_size": 100,
      "enabled": true
    },
    "cloudwatch_monitoring": {
      "enabled": true,
      "kinesis_alarms_enabled": true,
      "lambda_alarms_enabled": true,
      "settings": {
        "lambda_invocation_error_threshold": 0,
        "lambda_throttle_error_threshold": 0,
        "kinesis_iterator_age_error_threshold": 1000000,
        "kinesis_write_throughput_exceeded_threshold": 10
      }
    },
    "streamalert": {
      "classifier_config": {
        "enable_custom_metrics": true,
        "log_level": "debug",
        "log_retention_days": 14,
        "memory": 128,
        "timeout": 60
      }
    }
  },
  "region": "eu-north-1"
}
```

## Desired Change

A stable Terraform state, or if it is due to a configuration problem a warning/error should be displayed from `./manage.py build`

**Comment by jack1902** [comment]
Just been digging into this issue, and i believe i have found the root cause. Just going to see if can implement a fix but my belief is that these two modules:

- `terraform/modules/tf_monitoring`
- `terraform/modules/tf_lambda`

are stepping on each others toes. Here is my potential proof:

https://github.com/airbnb/streamalert/blob/d0e28f330a8636b9e080540ba098500801a857dc/terraform/modules/tf_lambda/cloudwatch.tf#L40-L59

https://github.com/airbnb/streamalert/blob/d0e28f330a8636b9e080540ba098500801a857dc/terraform/modules/tf_monitoring/main.tf#L5-L27

**Comment by jack1902** [comment]
this is definitely the issue, across StreamAlert, cloudwatch metric alarms are configured in numerous places but point to the same resources. This should be consolidated into one place. Options:

1. Only use `tf_monitoring` to setup all of the metric alarms across the board (this is undesirable as the config files are currently mapped to the modules)

2. Each resource created such as `lambda`, `kinesis` should have their relevant modules create the relevant metric alarms with a `boolean` option to enable/disable

**Comment by jack1902** [comment]
# Temp Solution
As a temp solution, you can change the below to `false`

https://github.com/airbnb/streamalert/blob/d0e28f330a8636b9e080540ba098500801a857dc/conf/clusters/prod.json#L53

**Comment by chunyong-lin** [comment]
@jack1902  and me chatted via slack. And Jack brought up an interesting idea that we probably should refactor cloudwatch metric alarms which are living in multiple places. Two options
1. Fully deprecate [tf_monitoring terraform module](https://github.com/airbnb/streamalert/tree/release-3-0-0/terraform/modules/tf_monitoring) and move metric alarms to related modules.
2. Consolidate all metric alarms to tf_monitoring terraform module.

We discussed internally as well, we would prefer the option 1 that will make code cleaner and more reasonable. But we have two lambda functions are not use `tf_lambda` terraform module, which means these two lambda functions will still need `tf_monitoring` for metric alarms setup. There two functions are
1. athena_partition_refresh function. This function will be gone if we merge #874 and we probably will do by the end of this quarter.
2. threat_intel_downloader function. Good news, there is PR #1074 up for review and it will migrate this function to use `tf_lambda` module!!

We probably can come up some good ideas to deprecation `tf_monitoring` module sooner. 🤞

**Comment by ryandeivert** [comment]
this should be fixed with #1217
